rust/sealevel/programs/interchain-gas-paymaster/src/accounts.rs

```diff
@@ -0,0 +1,298 @@
+//! Interchain gas paymaster accounts.
+
+use std::{cmp::Ordering, collections::HashMap};
+
+use access_control::AccessControl;
+use account_utils::{AccountData, DiscriminatorData, DiscriminatorPrefixed, SizedData};
+use borsh::{BorshDeserialize, BorshSerialize};
+use solana_program::{clock::Slot, program_error::ProgramError, pubkey::Pubkey};
+
+use hyperlane_core::{H256, U256};
+
+use crate::error::Error;
+
+/// The scale for token exchange rates, i.e. a token exchange rate of 1.0 is
+/// represented as 10^19.
+pub const TOKEN_EXCHANGE_RATE_SCALE: u128 = 10u128.pow(19);
```

Comment:
Worth adding a ticket to try using a library for exchange rate conversions, to avoid exposing these constants in our smart contract logic. This one was decent usage: https://crates.io/crates/rusty-money